### PR TITLE
Polling for new notifications 

### PR DESCRIPTION
**Pull request by samgst-amazon**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
Implement polling service to check for new notifications from a remote endpoint

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationPollingService.kt [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+                stream.bufferedReader().readText()
+            }
+
+            deserializeNotifications(content)
+        } catch (e: Exception) {
+            LOG.error(e) { "Failed to load notifications from bundled resources" }
+            null
+        }
+    }
+
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationPollingService.kt:158]: isnt this just the `RemoteResource` system

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationPollingService.kt [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            .userAgent("AWS Toolkit for JetBrains")
+            .connect { request ->
+                request.connection.headerFields["ETag"]?.firstOrNull() ?: ""
+            }
+
+    /**
+     * Emits telemetry metric for polling failures
+     */
+    private fun emitFailureMetric(exception: Exception?) {
+        // todo: add metric
```
Comment (manodnyab) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationPollingService.kt:134]: I think this already exists in common, we can add this metric

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationServiceInitializer.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.intellij.ide.util.RunOnceUtil
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.startup.ProjectActivity
+
+class NotificationServiceInitializer : ProjectActivity {
+
+    override suspend fun execute(project: Project) {
+        val service = ApplicationManager.getApplication().getService(NotificationPollingService::class.java)
```
Comment (manodnyab) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationServiceInitializer.kt:14]: can we use the getInstance method here?

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationServiceInitializer.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        RunOnceUtil.runOnceForApp(this::class.qualifiedName.toString()) {
```
Comment (manodnyab) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationServiceInitializer.kt:15]: Do we need this? Wont the startup activity anyway run just once?

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationPollingService.kt [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            { startPolling() },
+            pollingIntervalMs
+        )
+    }
+
+    /**
+     * Main polling function that checks for updates and downloads if necessary
+     * Returns the parsed notifications if successful, null otherwise
+     */
+    private fun pollForNotifications(): Boolean {
```
Comment (manodnyab) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationPollingService.kt:92]: can we add tests for this?

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationServiceInitializer.kt (hunk 4 above)
Comment (samgst-amazon) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationServiceInitializer.kt:15]: I added this to explicitly only run once per IDE instance and not per-project. It's what I understood to work, don't know about a better way yet

**Pull request closed without merging**